packages/app/src/game/module/chat/state.ts

```diff
@@ -0,0 +1,52 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { PlayerListPlayer, QuickChatValue } from "@creature-chess/models";
+
+export type QuickChat = {
+	id: string;
+	value: QuickChatValue;
+	receivedAt: number;
+
+};
+export type QuickChatAction = {
+	sendingPlayerId: string;
+	receivingPlayerId: string;
+	chatValue: QuickChatValue | null;
+};
+
+const initialState: QuickChat[] = [];
+
+export const { reducer, actions: commands } = createSlice({
+	name: "quickChat",
+	initialState,
+	reducers: {
+		setPlayerChat: (state, { payload: { sendingPlayerId, receivingPlayerId, chatValue } }: PayloadAction<QuickChatAction>) => {
+			const existingChat = state.find(chat => chat.id === sendingPlayerId);
+			if (existingChat) {
+				const newState = state.filter(chat => chat.id !== existingChat.id);
```

Comment:
(future improvement again) if this state was indexed by player id, you wouldn't even have to check here, you could just always

```ts
return {
    ...state,
    [sendingPlayerId]: { /* ... */ }
}
```